runner/execer/fake/intercept.go

```diff
@@ -0,0 +1,31 @@
+package fake
+
+import (
+	"github.com/scootdev/scoot/runner/execer"
+)
+
+type InterceptExecer struct {
+	Condition   func(execer.Command) bool
+	Interceptor execer.Execer
+	Default     execer.Execer
+}
+
+func (e *InterceptExecer) Exec(command execer.Command) (execer.Process, error) {
+	if e.Condition(command) {
+		return e.Interceptor.Exec(command)
+	} else {
+		return e.Default.Exec(command)
+	}
+}
+
+func StartsWithSimExecer(cmd execer.Command) bool {
+	return len(cmd.Argv) > 0 && cmd.Argv[0] == "#! sim execer"
```

Comment:
can we make this string a constant?  This way it can easily be referenced in say test cases or generators?